docs/rfcs/016-holdings_service/README.md

```diff
@@ -0,0 +1,128 @@
+teams outside of the digital platform, or adding more places for library members
+to manage their relationship with us (i.e. account information).
+
+The project will most likely be a cross team (experience / data / preservation)
+endeavour, so the process in how we communicate and share what we are doing will
+be important too.
+
+## **Proposed solution**
+
+[https://docs.google.com/drawings/d/1jAAiUS2A2wifiV2ZL5RkdPzHDHvOzTSQPE-HWwxeyMU/edit?usp=sharing](https://docs.google.com/drawings/d/1jAAiUS2A2wifiV2ZL5RkdPzHDHvOzTSQPE-HWwxeyMU/edit?usp=sharing)
+
+### Client (Experience)
+
+At first the client would make calls to a separate Holdings API and
+progressively enhance the page as we would not want to block rendering of the
+integral catalogue information, nor could we rely on the consistency of the data
+with Sierra, at least at first. See data principles below.
+
+### Identity (Preservation)
+
+Requests will need to be authenticated and identified through the current
+identity service, CAS, as we will need to update Sierra with requests to not
+disrupt the current workflow of library desk staff.
+
+This authentication can happen as the first step and hopefully we will be able
+to propegate a token through to the holdings API which will in turn us it to
+authenticate with the sierra integration.
+
+There is currently a process for setting up a new service with SSO where the
+keys and knowledge sit with Digirati.
+
+We need to make sure we don't get sidelined into recreating a whole identity
+management system as that is not in scope, but to integrate with what we have.
+
+**Questions**
+
+- Do we need to have a proxy API in front of this?
+- Do we just keep using the library application already setup?
+
+### Holdings API (Data)
+
+There will be a public API that reflects the service as we would like it to be
+going forward. This will also allow for rapid iteration on the API and client
+while other parts of the service are being worked on.
+
+It will also allow us to build on any missing parts that we arise through user
+research as quick wins.
+
+The design and implementation should also be in the shape of what the data team
+would like to set as the standard for APIs going forward, including logging,
+monitoring and reporting.
+
+**Questions**
+
+- Should we use AKKA HTTP?
+- What is acceptable times when waiting on a response from Sierra?
+
+### Sierra integration (Preservation)
+
+All requests will have to be updated within Sierra as this is the current system
+used by library desk staff and people using the library, including services such
+as "Items previously checked out", "Saved searches", "Saved Lists", and "My
+bookmarks".
+
+- Loads of questions around where this exists in the network?
+
+### Catalogue API integration (Data)
+
+Start to think and design about how this information might be reflected in the
+catalogue API. This might not be an integral part to getting something released,
+but would be useful to keep in mind.
```

Comment:
It don’t think it should be - availability/ordering is clearly a different service

Reply:
Two questions:

* Would we ever want to be able to query on availability?
* Would we consider the availability integral information or secondary? If it's not in the API, we will not be able to render on first page load.

The second there are always solutions to, the first needing way more thought if we'd like to do it.

If not, great, we can pull it out.

Reply:
I think probably not on the filtering. I also think it's fine to consider it secondary information.

Definitely interested in @kenoir's take on this too.

Reply:
As a first part of this piece of work it definitely wouldn't sync with the API.

There isn't an internal requirement to query on this data.

I can't see any opportunity that this might open to people exploring our data (although sometimes you don't know).

The one place I can see this potentially being an issue is on the results if we'd like to display the status of an object there (quite reasonable I would say). But there are solutions around this, when building out a discussion platform we would have an endpoint such as:

`/status?ids=12345,akjsdh6,fkhak8hj,...`

And we would then return a list of statuses to keep the request count down on the client and be able to do one re-render. I think it's still used so can't be that bad a solution.

Reply:
I can imagine wanting to filter works results by availability in a performant fashion - but that's just another data source fed into the catalogue pipeline. Not a significant design configuration right now I think.

Reply:
removed.